encord/objects/bitmask.py

```diff
@@ -112,6 +112,21 @@ def _mask_to_rle(mask: bytes) -> List[int]:
     return rle_counts
 
 
+def transpose_bytearray(byte_data: bytes, shape: Tuple[int, int]) -> bytearray:
```

Comment:
I am concerned that this will be very slow in python.

Reply:
```
from time import perf_counter

import numpy as np


def transpose_bytearray(byte_data: bytes, shape: tuple[int, int]) -> bytearray:
    """
    Transpose a 2D array represented by bytes.
    """
    rows, cols = shape
    # Create a new bytearray to hold the transposed data
    transposed_byte_data = bytearray(len(byte_data))
    # Transpose the 2D array
    for row in range(rows):
        for col in range(cols):
            transposed_byte_data[col * rows + row] = byte_data[row * cols + col]

    return transposed_byte_data


for size in [10, 20, 40, 80, 160, 320, 640, 1280, 2560, 5120]:
    arr = (np.random.normal(size=(size, size)) * 255).astype(np.uint8)

    b = arr.tobytes()
    t0 = perf_counter()
    b2 = transpose_bytearray(b, arr.shape)
    t1 = perf_counter()
    arr_T = arr.T
    t2 = perf_counter()
    arr2 = np.frombuffer(b2, dtype=arr.dtype).reshape(arr.shape)

    assert np.allclose(arr.T, arr2)
    print(
        f"{str(arr.shape)=:<25s} Our transpose: {t1 - t0:.4f} sec, numpy: {t2 - t1:.4f} sec"
    )
```

yields these time comparisons:
<img width="697" alt="image" src="https://github.com/encord-team/encord-client-python/assets/93145535/09995aeb-ba80-49e2-adb6-f0781dcef597">

Reply:
Also, it's probably worth mentioning that this transpose function will only work for 8bit formats. For, e.g., float32, it'll fail because we'd have to read in more bytes per entry. Just as a comment in the code I suppose.

Reply:
Yh I agree however this is necessary to not add numpy as an explicit dependency. This is the solution me and @sergei-encord hit upon to deal with Encords repr of RLE strings being different to COCOs. Technically we could include a try import and use this only as a fallback. Thoughts?

Reply:
another option would be to transpose the RLE list itself which will change the complexity from O(W*H) to O(W*log(H)), but it requires a data structure that may not be in the builtins or replicated with those. I may write something fast if you find it useful to compare against with the alternatives.

Reply:
If you think that you could do as such @eloy-encord without too much difficulty, then that would be good and very cool! I was wondering about transposing in the RLE but not obvious to me how to do it.